charts/redpanda/statefulset.go

```diff
@@ -847,3 +886,442 @@ func templateToContainers(dot *helmette.Dot, template string) []corev1.Container
 	result := helmette.Tpl(template, dot)
 	return helmette.UnmarshalYamlArray[corev1.Container](result)
 }
+
+func StatefulSet(dot *helmette.Dot) *appsv1.StatefulSet {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	if !RedpandaAtLeast_22_2_0(dot) && !values.Force {
+		sv := semver(dot)
+		panic(fmt.Sprintf("Error: The Redpanda version (%s) is no longer supported \nTo accept this risk, run the upgrade again adding `--force=true`\n", sv))
+	}
+	ss := &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "StatefulSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      Fullname(dot),
+			Namespace: dot.Release.Namespace,
+			Labels:    FullLabels(dot),
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: StatefulSetPodLabelsSelector(dot),
+			},
+			ServiceName:         ServiceName(dot),
+			Replicas:            ptr.To(values.Statefulset.Replicas),
+			UpdateStrategy:      helmette.UnmarshalInto[appsv1.StatefulSetUpdateStrategy](values.Statefulset.UpdateStrategy),
+			PodManagementPolicy: "Parallel",
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      StatefulSetPodLabels(dot),
+					Annotations: StatefulSetPodAnnotations(dot, statefulSetChecksumAnnotation(dot)),
+				},
+				Spec: corev1.PodSpec{
+					TerminationGracePeriodSeconds: ptr.To(values.Statefulset.TerminationGracePeriodSeconds),
+					SecurityContext:               PodSecurityContext(dot),
+					ServiceAccountName:            ServiceAccountName(dot),
+					ImagePullSecrets:              values.ImagePullSecrets,
+					InitContainers:                StatefulSetInitContainers(dot),
+					Containers:                    StatefulSetContainers(dot),
+					Volumes:                       StatefulSetVolumes(dot),
+					TopologySpreadConstraints:     nil, // set below
+					NodeSelector:                  statefulSetNodeSelectors(dot),
+					Affinity:                      statefulSetAffinity(dot),
+					PriorityClassName:             values.Statefulset.PriorityClassName,
+					Tolerations:                   statefulSetTolerations(dot),
+				},
+			},
+			VolumeClaimTemplates: nil, //TODO
+		},
+	}
+
+	// TopologySpreadConstraints
+	// Was protected with this: semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion
```

Comment:
I think it's fairly safe to drop support for 1.16. Turns out the minimum required version defined by Chart.yaml is 1.21. So very safe to drop actually.

Reply:
In which case I'l drop the "Capabilities" bit :-)

Reply:
(done)